tensorflow_probability/python/util/deferred_tensor_test.py

```diff
@@ -335,6 +335,17 @@ def testOperatorIter(self):
         for _ in iter(x):
           pass
 
+  def testMethodNumpy(self):
```

Comment:
Is this the right place for the test?

Reply:
Yeah this is a good place. I'd suggest something like the following here:

```python
x_ = np.array([0., 1.])
x = tfp.util.DeferredTensor(tf.Variable(x_), tf.math.exp)

if tf.executing_eagerly():
  self.assertAllEqual(np.exp(x_), x.numpy())
else:
  with self.assertRaises(...):
    x.numpy()
```

As you probably figured out, we generate a few different test cases for each test method (this is due to the `@test_all_tf_execution_regimes` decorator):
  1. graph mode
  2. eager mode
  3. eager mode with all tf.function wrappers disabled (these can sometimes be generated by library code, and we want to ensure that a user debugging something sees consistent behavior with `tf.function`s enabled and disabled).

As for step-through debugging tests, there might be some helpful hints [here](https://stackoverflow.com/questions/51709590/can-i-use-python-debugger-in-bazel-test) but personally, after 20+ years, I still just use print statements for debugging 😝 ...

Reply:
Thanks for the comments! I fixed the graph-mode part of the test. 

> I still just use print statements for debugging

Haha, that's definitely not a bad way to go 😄

Reply:
Just want to make sure, since it is a bit subtle in the UI, that the "here" in my comment was a link to this SO post, which may be helpful: https://stackoverflow.com/questions/51709590/can-i-use-python-debugger-in-bazel-test

Reply:
Oh yeah, I figured. I tried the `--run_under` examples (both with `--run_under=.../bin/ipdb3` and `--run_under=.../lib/python3.7/pdb.py`) from the SO post, but for some reason neither stops in to the debugger but it results in this weird behavior where the test passes no matter what I do to it (even if there's a syntax error and it fails without `--run_under`). Not sure if that's due to caching or something. I'll try to figure this out again next time I have to run the tests 🙂

Reply:
Ah, sorry about that. 😬If you ever figure it out let us know and we can add some hints to the contributor docs!